### PR TITLE
Add script to produce DOT for funcs in a CPG

### PR DESCRIPTION
**Pull request by GlassAndOneHalf**
- Adds a new script, `functions-to-dot`, that produces a DOT graph for all internal functions contained within a CPG.

### CODE REVIEW

**Comment by GlassAndOneHalf** [comment]
I was thinking it might be nice to add a new function to `Method` on `codepropertygraph` that allows a user to generate a graph for any arbitrary method, e.g.
`cpg.method.dotGraph` or `cpg.method.name(".*add.*").dotGraph`

File: scripts/functions-to-dot/description.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "name": "functions-to-dot",
+  "description": "Generates a dot graph for each function in the provided CPG. Each file will contain"
+}
```
Comment (max-leuthaeuser) [inline scripts/functions-to-dot/description.json]: Something is missing here.

File: scripts/functions-to-dot/description.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "name": "functions-to-dot",
+  "description": "Generates a dot graph for each function in the provided CPG. Each file will contain"
```
Comment (fabsx00) [inline scripts/functions-to-dot/description.json]: Sentence incomplete

File: scripts/functions-to-dot/functions-to-dot.scala [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+def stringifyDotFromMethod(method: Method): String = {
+  val sb = new StringBuilder
+  sb.append(s"digraph ${method.name} {\n node[shape=box];\n")
+  sb.append(dotFromMethod(method).mkString("\n"))
+  sb.append("\n}\n")
+  sb.toString
+}
+
+cpg.method.filterNot(_.external).l.map { method =>
```
Comment (max-leuthaeuser) [inline scripts/functions-to-dot/functions-to-dot.scala]: I thing there is already an `internal` traversal.

File: scripts/functions-to-dot/functions-to-dot.scala [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+import org.apache.tinkerpop.gremlin.structure.{ Direction, Vertex }
+
+import io.shiftleft.codepropertygraph.generated._
+import io.shiftleft.codepropertygraph.generated.nodes.Expression
+import java.nio.file.Paths
+import java.net.URI
+
+import scala.annotation.tailrec
+
+def getTopLevelExpressions(expression: Vertex): List[Expression] = {
```
Comment (fabsx00) [inline scripts/functions-to-dot/functions-to-dot.scala]: Let's see if we can make query language steps from some of these.

Comment (fabsx00) [reply scripts/functions-to-dot/functions-to-dot.scala]: (after merging, that is)

**Pull request merged**